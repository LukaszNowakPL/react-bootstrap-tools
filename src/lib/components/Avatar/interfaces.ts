export interface ConfigInterface {
  defaultIcon: string;
  defaultWrapperClassName: string;
  defaultImageClassName: string;
  defaultIconClassName: string;
}

export interface AvatarProps {
  src?: string;
  icon?: string;
  className?: string;
  imageClassName?: string;
  iconClassName?: string;
}

export interface WrapperProps {
  className: string;
}

export interface ImageProps {
  src: string;
  fluid: true;
  className: string;
}

export interface IconProps {
  icon: string;
  className: string;
}
