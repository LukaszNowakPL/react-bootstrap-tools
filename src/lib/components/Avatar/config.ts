import { ConfigInterface } from "./interfaces";

const config: ConfigInterface = {
  defaultIcon: "person",
  defaultWrapperClassName: "Avatar",
  defaultImageClassName: "Avatar__image",
  defaultIconClassName: "Avatar__icon"
};

export default config;
