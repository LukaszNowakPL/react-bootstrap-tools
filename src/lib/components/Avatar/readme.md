# Avatar

Component for showing user avatar.

![Avatar example](../../../../readme_img/AvatarExample.png)

## Example of usage

```
import React from 'react';
import { Avatar } from 'react-bootstrap-tools';

const SimplestExample = () => (
    <Avatar />
);

export default SimplestExample;
```

## Available optional props

- string `src` : Url of a picture to use. If not provided material icon will show instead.
  Additional props connected with `src` prop:
  - string `imageClassName` : Additional class name to custom your styling of provided image
- string `icon` : Id of a material icon to show in case of no `src` prop provided. This id will override the default icon. Click [here](https://material.io/tools/icons/?style=baseline) to check available icons library.
  Additional props connected with `withIcon` prop:
  - string `iconClassName` : Additional class name to custom your styling of provided icon
- string `className` : Additional class name to custom your styling of a component. 
