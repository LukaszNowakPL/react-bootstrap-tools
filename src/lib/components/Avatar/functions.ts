import { AvatarProps, IconProps, ImageProps, WrapperProps } from "./interfaces";
import config from "./config";

export const getWrapperProps = (conf: AvatarProps) => {
  const { className } = conf;

  const wrapperProps: WrapperProps = {
    className:
      config.defaultWrapperClassName + (className ? " " + className : "")
  };

  return wrapperProps;
};

export const getImageProps = (conf: AvatarProps) => {
  const { src, imageClassName } = conf;

  const imageProps: ImageProps = {
    fluid: true,
    src: src || "",
    className:
      config.defaultImageClassName +
      (imageClassName ? " " + imageClassName : "")
  };

  return imageProps;
};

export const getIconProps = (conf: AvatarProps) => {
  const { icon, iconClassName } = conf;

  const iconProps: IconProps = {
    icon: icon ? icon : config.defaultIcon,
    className:
      config.defaultIconClassName + (iconClassName ? " " + iconClassName : "")
  };

  return iconProps;
};
