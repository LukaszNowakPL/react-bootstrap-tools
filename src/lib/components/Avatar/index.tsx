import * as React from "react";
import { PureComponent } from "react";
import { AvatarProps, IconProps, ImageProps, WrapperProps } from "./interfaces";
import { getIconProps, getImageProps, getWrapperProps } from "./functions";
import { Image } from "react-bootstrap";
import Glyphicon from "../Glyphicon";

export const AvatarImage = (props: AvatarProps) => {
  const imageProps: ImageProps = getImageProps(props);
  return <Image {...imageProps} />;
};

export const AvatarIcon = (props: AvatarProps) => {
  const iconProps: IconProps = getIconProps(props);
  return <Glyphicon {...iconProps} />;
};

class Avatar extends PureComponent<AvatarProps> {
  displayName: string = "Avatar";

  render() {
    const { src } = this.props;

    const avatarObject = !!src ? (
      <AvatarImage {...this.props} />
    ) : (
      <AvatarIcon {...this.props} />
    );

    const wrapperProps: WrapperProps = getWrapperProps(this.props);

    return <section {...wrapperProps}>{avatarObject}</section>;
  }
}

export default Avatar;
