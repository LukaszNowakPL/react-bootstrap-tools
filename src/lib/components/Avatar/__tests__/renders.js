import React from "react";
import { shallow } from "enzyme";
import Avatar, { AvatarIcon, AvatarImage } from "../index";

describe("<Avatar />", () => {
  const customComponentProps = {};

  it("renders", () => {
    const component = shallow(<Avatar {...customComponentProps} />);
    expect(component).toMatchSnapshot();
  });

  it("renders provided icon", () => {
    const component = shallow(
      <Avatar {...customComponentProps} icon="testIcon" />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders provided icon and icon's class name", () => {
    const component = shallow(
      <Avatar
        {...customComponentProps}
        icon="testIcon"
        iconClassName="testIconClassName"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders provided src", () => {
    const component = shallow(
      <Avatar {...customComponentProps} src="testSrc" />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders provided src and image's class name", () => {
    const component = shallow(
      <Avatar
        {...customComponentProps}
        src="testSrc"
        imageClassName="testImageClassName"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders with provided class name", () => {
    const component = shallow(
      <Avatar {...customComponentProps} className="testClassName" />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders with no icon if src is provided", () => {
    const component = shallow(
      <Avatar {...customComponentProps} icon="testIcon" src="testSrc" />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("<AvatarIcon />", () => {
  const customComponentProps = {};

  it("renders", () => {
    const component = shallow(<AvatarIcon {...customComponentProps} />);
    expect(component).toMatchSnapshot();
  });

  it("renders with provided icon", () => {
    const component = shallow(
      <AvatarIcon {...customComponentProps} icon="testIcon" />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders with provided iconClassName", () => {
    const component = shallow(
      <AvatarIcon {...customComponentProps} iconClassName="testIconClassName" />
    );
    expect(component).toMatchSnapshot();
  });
});

describe("<AvatarImage />", () => {
  const customComponentProps = {
    src: "testSrc"
  };

  it("renders", () => {
    const component = shallow(<AvatarImage {...customComponentProps} />);
    expect(component).toMatchSnapshot();
  });

  it("renders with provided imageClassName", () => {
    const component = shallow(
      <AvatarImage
        {...customComponentProps}
        imageClassName="testImageClassName"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
