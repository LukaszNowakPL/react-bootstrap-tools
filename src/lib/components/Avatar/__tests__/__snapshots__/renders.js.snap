// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders 1`] = `
<section
  className="Avatar"
>
  <AvatarIcon />
</section>
`;

exports[`<Avatar /> renders provided icon 1`] = `
<section
  className="Avatar"
>
  <AvatarIcon
    icon="testIcon"
  />
</section>
`;

exports[`<Avatar /> renders provided icon and icon's class name 1`] = `
<section
  className="Avatar"
>
  <AvatarIcon
    icon="testIcon"
    iconClassName="testIconClassName"
  />
</section>
`;

exports[`<Avatar /> renders provided src 1`] = `
<section
  className="Avatar"
>
  <AvatarImage
    src="testSrc"
  />
</section>
`;

exports[`<Avatar /> renders provided src and image's class name 1`] = `
<section
  className="Avatar"
>
  <AvatarImage
    imageClassName="testImageClassName"
    src="testSrc"
  />
</section>
`;

exports[`<Avatar /> renders with no icon if src is provided 1`] = `
<section
  className="Avatar"
>
  <AvatarImage
    icon="testIcon"
    src="testSrc"
  />
</section>
`;

exports[`<Avatar /> renders with provided class name 1`] = `
<section
  className="Avatar testClassName"
>
  <AvatarIcon
    className="testClassName"
  />
</section>
`;

exports[`<AvatarIcon /> renders 1`] = `
<Glyphicon
  className="Avatar__icon"
  icon="person"
/>
`;

exports[`<AvatarIcon /> renders with provided icon 1`] = `
<Glyphicon
  className="Avatar__icon"
  icon="testIcon"
/>
`;

exports[`<AvatarIcon /> renders with provided iconClassName 1`] = `
<Glyphicon
  className="Avatar__icon testIconClassName"
  icon="person"
/>
`;

exports[`<AvatarImage /> renders 1`] = `
<Bootstrap(Image)
  className="Avatar__image"
  fluid={true}
  src="testSrc"
/>
`;

exports[`<AvatarImage /> renders with provided imageClassName 1`] = `
<Bootstrap(Image)
  className="Avatar__image testImageClassName"
  fluid={true}
  src="testSrc"
/>
`;
